src/hooks/useWindowEventListener.ts

```diff
@@ -7,14 +7,21 @@ import { useGlobalObjectEventListener } from "./useGlobalObjectEventListener";
  *
  * @param {string} eventName The event to track
  * @param {Function} callback The callback to be called on event
- * @param {object} conditions The options to be passed to the event listener
+ * @param {boolean | {once?: boolean; passive?: boolean; signal?: AbortSignal; capture?: boolean;}} listenerOptions The options to be passed to the event listener
  * @param {boolean} isLayoutEffect Should it use layout effect. Defaults to false
  * @returns {undefined}
  */
 function useWindowEventListener(
   eventName: string,
   callback: (...args: any) => void,
-  listenerOptions: any = {},
+  listenerOptions:
```

Comment:
Since this is less readable, what do you think of creating a type in `src/types/utils.ts` and then refer that here? 

So perhaps something like 

```
type EventListernerOptions =  boolean
    | {
        capture?: boolean;
        once?: boolean;
        passive?: boolean;
        signal?: AbortSignal;
      } = {},
```

And the import it here?

Reply:
I thought about that, thanks for pointing out!
Sounds great, will do it!

Reply:
I wonder if we can refer to the same type in jsdoc comment easily from src/types/utils 🤔 .

Reply:
Seems `EventListenerOptions` is declared in typescript, so I'll name it `ListenerOptions`.

Reply:
The GitHub did not update the chat and I didn't saw your comment.
Do you mean to put the type in src/types/utils?

Reply:
Sounds great!
I just didn't know the directory structure.

Reply:
![image](https://user-images.githubusercontent.com/2936644/176672806-ca272eb0-db18-4764-9d31-918959961b86.png)
This one. This is a new file that I added today. You might want to pull `main` and rebase.

Reply:
Got it!